backend/tests/integration/tests/permissions/test_cc_pair_permissions.py

```diff
@@ -0,0 +1,165 @@
+    # Creating an admin user (first user created is automatically an admin)
+    admin_user: TestUser = UserManager.create(name="admin_user")
+
+    # Creating a curator
+    curator: TestUser = UserManager.create(name="curator")
+
+    # Creating a user group
+    user_group_1 = UserGroupManager.create(
+        name="curated_user_group",
+        user_ids=[curator.id],
+        cc_pair_ids=[],
+        user_performing_action=admin_user,
+    )
+    UserGroupManager.wait_for_sync(
+        user_groups_to_check=[user_group_1], user_performing_action=admin_user
+    )
+    # setting the user as a curator for the user group
+    UserGroupManager.set_user_to_curator(
+        test_user_group=user_group_1,
+        user_to_set_as_curator=curator,
+        user_performing_action=admin_user,
+    )
+
+    # Creating another user group that the user is not a curator of
+    user_group_2 = UserGroupManager.create(
+        name="uncurated_user_group",
+        user_ids=[curator.id],
+        cc_pair_ids=[],
+        user_performing_action=admin_user,
+    )
+    UserGroupManager.wait_for_sync(
+        user_groups_to_check=[user_group_1], user_performing_action=admin_user
+    )
+
+    # Create a credentials that the curator is and is not curator of
+    connector_1 = ConnectorManager.create(
+        name="curator_owned_connector",
+        source=DocumentSource.CONFLUENCE,
+        groups=[user_group_1.id],
+        is_public=False,
+        user_performing_action=admin_user,
+    )
+    # currently we dont enforce permissions at the connector level
+    # pending cc_pair -> connector rework
+    # connector_2 = ConnectorManager.create(
+    #     name="curator_visible_connector",
+    #     source=DocumentSource.CONFLUENCE,
+    #     groups=[user_group_2.id],
+    #     is_public=False,
+    #     user_performing_action=admin_user,
+    # )
+    credential_1 = CredentialManager.create(
+        name="curator_owned_credential",
+        source=DocumentSource.CONFLUENCE,
+        groups=[user_group_1.id],
+        curator_public=False,
+        user_performing_action=admin_user,
+    )
+    credential_2 = CredentialManager.create(
+        name="curator_visible_credential",
+        source=DocumentSource.CONFLUENCE,
+        groups=[user_group_2.id],
+        curator_public=False,
+        user_performing_action=admin_user,
+    )
+
+    # END OF HAPPY PATH
+
+    """
+    Curators should not be able to:
+    - Create a public cc pair
+    - Create a cc pair for a user group they are not a curator of
+    - Create a cc pair for a user group that the connector does not belong to (NOT WORKING)
+    - Create a cc pair for a user group that the credential does not belong to
+    """
+    with pytest.raises(HTTPError):
+        CCPairManager.create(
+            connector_id=connector_1.id,
+            credential_id=credential_1.id,
+            name="invalid_cc_pair_1",
+            groups=[user_group_1.id],
```

Comment:
could we remove this group here? This would make this case simpler remove other reasons why this would fail (e.g. if we later change this to make the user not a curator of `user_group_1` but forget to change this, it would no longer test what it is supposed to test)

Reply:
no, then it would fail because curator is only a curator of group 1. if we dont designate user_group_1, it would be invalid because curators have to attach a cc_pair to a group to create a cc_pair

Reply:
That being said i can add a test for explicity trying to create a cc pair without any groups